### DISCUSSION TITLE
How to print invoice / export pdf-excel

### DISCUSSION THREAD

**Discussion by viayuve** [sqlpage/SQLPage#640] [Q&A]
Hello Guys
Question - how can I print invoice? or Export pdf-excel file 
What I have tried - design invoice - too much spacing in between components - how to reduce it?

If anyone have built something similar - links examples

I am new here and almost week before I started doing it with nick's tuts https://learnsqlpage.com/

**Comment by lovasoa** [comment]
Hello and welcome to SQLPage !
  Can you tell us more about how you found SQLPage and what you are building ?
  
  About the print button, you can see this previous discussion: https://github.com/lovasoa/SQLpage/discussions/499 . In short, it is possible, but not built into SQLPage. You have to write a small bit of javascript code yourself.
  You excel exports, you can use the csv component if you only have textual data: https://sql.datapage.app/component.sql?component=csv
  If you want an excel-like experience in the browser, you can use the SQLPage spreadsheet component: https://github.com/lovasoa/sqlpage-spreadsheet/

**Reply by viayuve** [reply]
Finding process

Not a developer but finding my way @lovasoa I wanted solution like single exe and single language solution, where I found "rebol" after that rebolforum.com by nick. In his forum I found https://learnsqlpage.com/ - SQL language I know bit , python not at all anvil not my cup of tea. so I am here. I do run small sticker printing business from home. Sunday I was free tried nicks lot of tuts was so easy and chatgpt helped a lot. 

What am I building.

Small App to make my life easy.
Inward/Outward - of stickers roll - Stock management. - inward done working on outward.
Create Invoice/Challan for client. Invoice management.

Thank you I will look into your suggestion.

**Reply by alexisrc1** [reply]
Hello @viayuve and welcome to SQLPage, I remain at your disposal if you have any questions !
Alexis
